src/components/badge/StyledIcon.ts

```diff
@@ -0,0 +1,69 @@
+import { styled } from 'emotion-solid'
+import type { JSX } from 'solid-js'
+import tw from 'twin.macro'
+
+import type { BadgeProps } from '.'
+
+type StyledIconProps = BadgeProps & JSX.IntrinsicElements['span']
+
+const baseStyles = [
+  tw`absolute flex h-4 w-4 bg-blue justify-center items-center text-xxs font-sans border border-white rounded-full`,
```

Comment:
Might need to add a z index here, when I tried to use the button component, the badge was hidden behind the button.